Linked subsystems.

VytalyGorbatov opened the pull request:
Dev branch is a branch where Services, Sockets, Logger and ThreadPool is merged and working fine. Also MVS19 solution is configured quite well, so work with this branch is much more convenient.

VytalyGorbatov committed:
Cleaned directories

VytalyGorbatov committed:
MVS19 Solution UPD

Chupakabra0 committed:
Update CLogMacro.h

Chupakabra0 committed:
Update: now param print outs variable type

Serwlashev committed:
Added possibility to connect to the server multiple times

VytalyGorbatov committed:
Linked app with Service branch && Fixed mess with Sockets dir

VytalyGorbatov committed:
Delete Sockets/BackEndMonitoringSockets/Build directory

VytalyGorbatov committed:
Update .gitignore

Serwlashev committed:
Changes before marging with dev branch

Serwlashev committed:
Solved merge conflicts with dev branch

Chupakabra0 committed:
Merge remote-tracking branch 'origin/Dev' into logger

Chupakabra0 committed:
Update: removed redundant std::stringstream from PrintLogMessage()

Chupakabra0 committed:
Added new time get function

Chupakabra0 committed:
Fix: now runtime-time prints correctly

Chupakabra0 committed:
Working libs update

Chupakabra0 committed:
Update: refactoring of some macro-functions.

VytalyGorbatov committed:
Moved thread pool initialization from Sockets

Chupakabra0 committed:
Update: all utils now inline functions with xml-doc

VytalyGorbatov committed:
Fixed error with precompiled headers

AntonVoytov commented on Sockets/BackEndMonitoringSockets/PropertySheets/IncludeSheetRelease64.props:
```diff
@@ -0,0 +1,13 @@
+  <ItemDefinitionGroup>
+    <Link>
+      <AdditionalLibraryDirectories>$(SolutionDir)packages\CLogger\$(Configuration)\x64\; %(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalDependencies>CLogger.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
```
Looks like there is too many lib dependencies, some of them are not might needed. Please check other places.

VytalyGorbatov replied:
Yes, sorry, this was miscopied. Edited it in [commit](https://github.com/ITA-Dnipro/BackEndMonitoring/pull/28/commits/74fd40f3cec3740749b2632ea0a8e42f7e3dc2df).

AntonVoytov reviewed (commented):
Please check comment with lib dependencies.

VytalyGorbatov committed:
Edited Sockets property sheets

Chupakabra0 committed:
Update: more simple and universal time get function

Chupakabra0 committed:
New version of libs

Chupakabra0 committed:
Fix: now log with params and vars prints correctly

Chupakabra0 committed:
Update CLogMessage.h

Chupakabra0 commented on Server/BackEndMonitoringServer/src/BackEndMonitoringServer.cpp:
```diff
@@ -1,27 +1,23 @@
-	system("pause");
-	return 0;
-}
+    const int return_code = success ? 0 : 1;
```
If `!success` is `true`, then you step into if statement and `EXIT_FALIURE` will be returned. So you'll never get `return_code` = 1 in this situation. You can safely return `EXIT_SUCCESS`.

VytalyGorbatov replied:
Thnx, fixed.

Chupakabra0 commented on Server/BackEndMonitoringServer/src/CServiceHandler.cpp:
```diff
@@ -0,0 +1,287 @@
+#include "CServiceHandler.h"
+#include "Utils.h"
+
+class ServiceHandle
```
Don't forget to decide what you have to do with move and copy ctors in future. For example, you can safely delete copy and leave default move ctors in this class.

VytalyGorbatov replied:
Fixed too.

Serwlashev committed:
Deleted CAcceptorSocket and CConnectorSocket, replaced all specific platform code to the PlatformUtils, deleted all unimportant features after changing sockets, replaced almost all header to the cpp files

Serwlashev committed:
Deleted all unnecessary garbage from the branch

Serwlashev committed:
Fixed problem with deleting dll libraris and following problems with building

Crowy-o commented on Server/BackEndMonitoringServer/include/CService.h:
```diff
@@ -0,0 +1,64 @@
+
+struct ServiceParameters
+{
+    const CString& name         = "BackendMonitoringService";
```
we don't need such formatted lines according to code style

VytalyGorbatov replied:
Fixed, thnx.

Crowy-o commented on Server/BackEndMonitoringServer/include/CService.h:
```diff
@@ -0,0 +1,64 @@
+    SERVICE_STATUS_HANDLE m_status_handle;
+    std::thread m_main_thread;
+    CEvent m_stop_event;
+    std::shared_ptr<CThreadPool> m_thread_pool;
```
it's pointer. Must be m_p_thread_pool

VytalyGorbatov replied:
Fixed.

Crowy-o commented on Server/BackEndMonitoringServer/include/Utils.h:
```diff
@@ -0,0 +1,10 @@
+
+namespace Utils
+{
+
```
don't need empty line

VytalyGorbatov replied:
This files useless, deleted now.

Crowy-o commented on Server/BackEndMonitoringServer/src/CCommandLineHandler.cpp:
```diff
@@ -0,0 +1,53 @@
+
+	    if (argc == 2)
+        {
+            if (strcmp(argv[1], "install") == 0)
+            {
+                success = service_handler->Install();
+                break;
+            }
+
+            if (strcmp(argv[1], "uninstall") == 0)
+            {
+                success = service_handler->Uninstall();
+                break;
+            }
+
+            if (strcmp(argv[1], "help") == 0)
+            {
+                // Show help
+                break;
+            }
```
You could put if -> else if -> else - then if made a mistake, help command will be shown

AntonVoytov replied:
The better option is to change 'if else' on 'switch'

VytalyGorbatov replied:
Thnx, changed to switch-case

Crowy-o commented on Server/BackEndMonitoringServer/src/CService.cpp:
```diff
@@ -0,0 +1,149 @@
+        ELogConfig::LINE_NUMBER, ELogConfig::MESSAGE, ELogConfig::PARAMS);
+    auto logger = builder.BuildSharedLog();
+
+    size_t num_threads = 20;
+    int port = 1111;
+    std::string ip_address = "127.0.0.1";
```
maybe better to write comment that here will be XMLParser used?

VytalyGorbatov replied:
Added \\Todo

Crowy-o commented on Server/BackEndMonitoringServer/src/CService.cpp:
```diff
@@ -0,0 +1,149 @@
+    return ::StartServiceCtrlDispatcher(table_entry) == TRUE;
+}
+
+const CString& CService::GetName() const 
+{ 
+    return m_name; 
+}
+
+const CString& CService::GetDisplayName() const 
+{ 
+    return m_display_name;
+}
+
+const DWORD CService::GetStartType() const
+{ 
+    return m_start_type;
+}
+
+const DWORD CService::GetErrorControlType() const
+{
+    return m_error_control_type;
+}
```
formate to { return ...; }

VytalyGorbatov replied:
Formatted

Crowy-o commented on Server/BackEndMonitoringServer/src/CServiceHandler.cpp:
```diff
@@ -0,0 +1,287 @@
+
+    ~ServiceHandle() 
+    {
+        if (m_handle != nullptr) 
+        {
+            if(::CloseServiceHandle(m_handle) == 0)
```
swap places: nullptr != ....

VytalyGorbatov replied:
Edited

Crowy-o commented on Server/BackEndMonitoringServer/src/CServiceHandler.cpp:
```diff
@@ -0,0 +1,287 @@
+        }
+
+        escaped_path.ReleaseBuffer();
+        escaped_path.Remove('\"');
+
+        escaped_path = '\"' + escaped_path + '\"';
```
maybe better write fucntion FormatePath(string)?

VytalyGorbatov replied:
Ok, @Serwlashev please, edit this in Sockets branch.

Crowy-o commented on Server/BackEndMonitoringServer/src/CServiceHandler.cpp:
```diff
@@ -0,0 +1,287 @@
+
+bool ServiceHandler::Run() const
+{
+    return m_service->Run();
```
I couldn't find header, but, as I see, m_service - pointer. Then rename it into m_p_service

VytalyGorbatov replied:
Fixed

Crowy-o commented on Sockets/BackEndMonitoringSockets/include/CAcceptorWrapper.h:
```diff
@@ -10,18 +10,18 @@ class CAcceptorWrapper
 
 	CEvent& m_event;
-	std::unique_ptr<CThreadPool> m_pool;
+	std::shared_ptr<CThreadPool> m_pool;
 	std::unique_ptr<CAcceptor> m_server_acceptor;
 	std::unique_ptr<CServiceConnectionHandler> m_service_handler;
 	std::unique_ptr<CSocketWrapper> m_stream;
```
if pointer - add prefix p_ (example: m_p_pool)

VytalyGorbatov replied:
@Serwlashev, this too.

Crowy-o commented on Sockets/BackEndMonitoringSockets/src/CAcceptor.cpp:
```diff
@@ -18,6 +18,11 @@ int CAcceptor::GetHandle() const
 }
 
+bool CAcceptor::CloseSocket()
+{
+	return m_peer_acceptor->CloseSocket();
+}
```
formate as 1 line { ..... }

VytalyGorbatov replied:
@Serwlashev, also this)

Crowy-o commented on Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp:
```diff
@@ -2,38 +2,40 @@
+	std::shared_ptr<CLogger> logger)
+	: m_event(event), m_pool(pool), m_logger(logger)
 {
 	m_server_acceptor = InitAcceptor(1111, "127.0.0.1");
 	m_stream = InitSocketWrapper(m_server_acceptor->GetHandle());
 	m_service_handler = InitServiceHandler(m_server_acceptor->GetHandle());
```
The constructor shouldn't implement hard init. Better to add another method for such initialization and added variable that check is_initialized object

VytalyGorbatov replied:
@Serwlashev, and this

Chupakabra0 committed:
Update: C-style time get

Chupakabra0 committed:
Update: renaming of macros

Chupakabra0 committed:
Update: added new macro-functions for start\end of functions

Chupakabra0 committed:
Fix: corrected macros' names in macros

Chupakabra0 committed:
Fix: strange bug with zero-size std::string

Chupakabra0 committed:
Update CLogger.vcxproj

Chupakabra0 committed:
Fix: clean redundant symbols in time

Chupakabra0 committed:
Fix: missed symbols in macros

Chupakabra0 committed:
Update: more informatic message

VytalyGorbatov committed:
Fixed according to pull request #28

Chupakabra0 committed:
Update: added global extern vars, that macros use

Chupakabra0 committed:
Fix: moved extern vars to headers

Chupakabra0 committed:
Updated: reworking and adding some macros

Chupakabra0 committed:
Update: now message contains quots

Chupakabra0 committed:
Update: added flush|not-flush logger feature

Chupakabra0 committed:
Removed utils from src

Chupakabra0 committed:
Merge remote-tracking branch 'origin/Dev' into logger

Chupakabra0 committed:
Update: now logger uses project utils

Chupakabra0 committed:
Update: added log flush for builder

Chupakabra0 committed:
Update: new macro comand for log flush

Chupakabra0 committed:
A lot fixes to make this **** work

AntonVoytov commented on Server/BackEndMonitoringServer/include/CService.h:
```diff
@@ -5,11 +5,11 @@
 struct ServiceParameters
 {
-    const CString& name         = "BackendMonitoringService";
+    const CString& name = "BackendMonitoringService";
```
Replace all CString with std::string.
const CString& name = "BackendMonitoringService"; ->  const std::string name("BackendMonitoringService"); if possible or init via ctor.
Check other places.

AntonVoytov commented on Server/BackEndMonitoringServer/src/CService.cpp:
```diff
@@ -62,31 +63,31 @@ DWORD WINAPI CService::ServiceCtrlHandler(
 void WINAPI CService::SvcMain(DWORD argc, CHAR** argv) 
 {
-    assert(m_service);
+    assert(m_p_service);
```
Do not assert, use conditional instead.

AntonVoytov commented on Server/BackEndMonitoringServer/src/CService.cpp:
```diff
@@ -62,31 +63,31 @@ DWORD WINAPI CService::ServiceCtrlHandler(
-    m_service = this;
+    m_p_service = this;
 
     CHAR* name = const_cast<CString&>(m_name).GetBuffer();
```
CHAR* -> char*

AntonVoytov reviewed (commented):
Please fix and reply to each comment.

VytalyGorbatov committed:
Hardware status monitoring -> Dev (#32)

* Merged HardwareMonitoring branch with Dev branch
Co-authored-by: Crowy-o <63596025+Crowy-o@users.noreply.github.com>

Serwlashev committed:
Fixed sending data to multiple clients (now they get response, but server with some reasons get wrong requests), fixed fields order in headers, now files before sending get header, which receive function reads and than create buffer according to it

Chupakabra0 committed:
Merge pull request #26 from ITA-Dnipro/logger

Small update of logger

Serwlashev committed:
Solved conflicts with logger branch

AntonVoytov commented on Server/BackEndMonitoringServer/src/CProcessInfo.cpp:
```diff
@@ -0,0 +1,173 @@
+bool CProcessInfo::CountCpuUsage()
+{
+    unsigned long long cur_sys_time, cur_kernel_time, cur_user_time;
+    cur_sys_time = cur_kernel_time = cur_user_time = 0;
```
Do not do that! It looks disgusting. Initialize simple types when declare.

AntonVoytov commented on utils/include/WindowsUtils.cpp:
```diff
@@ -0,0 +1,113 @@
+	bool GetExistingProcessIds(std::vector<unsigned>& container_of_PIDs)
+	{
+		unsigned short m_max_process_count = 1024;
+		DWORD* p_process_ids = new DWORD[m_max_process_count];
```
Try to use a smart pointer for a array

AntonVoytov commented on utils/include/WindowsUtils.cpp:
```diff
@@ -0,0 +1,113 @@
+		HANDLE process = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ,
+			FALSE, PID);
+
+		bool success = (process != 0);
```
Check what HANDLE type is actually. It should be void* or void**

AntonVoytov commented on utils/include/Utils.h:
```diff
@@ -1,16 +1,28 @@
 	/// <returns>format: dd.mm.yyyy hh:mm:ss</returns>
-	bool GetCurrentDateAndTimeFormatted(std::string& 
-		date_time_var_to_save);
+	[[nodiscard]] bool TryGetCurrentDateAndTimeFormatted(std::string&
```
If you prefer to use a single return, try to use do {} while construction.

AntonVoytov reviewed (commented):
Please check comments

Chupakabra0 committed:
Fix: now return works correctly

Chupakabra0 committed:
Fix: now param macros works correctly

Chupakabra0 committed:
New .dll versions uploaded to branch

Chupakabra0 committed:
Merge branch 'logger' into Sockets

Chupakabra0 committed:
Integrating logger in sockets

Chupakabra0 committed:
Fix: now logger correctly works in multiple files

Chupakabra0 committed:
Merge branch 'logger' into Sockets

Chupakabra0 committed:
Urgent building fix

Chupakabra0 committed:
Added new preprocessor for correct work of logger

Serwlashev committed:
Solved problems with compilation after merging with logger

Serwlashev committed:
Fixed problems with connection multiple clients to the server (Now it really works), added logger to the most important operations

Chupakabra0 committed:
Merge branch 'Dev' into logger

Chupakabra0 committed:
Fix: now it returns again

Chupakabra0 committed:
New libs

Serwlashev committed:
solved problems with clients disconnection, now new clients can be added to the thread pool

Serwlashev committed:
Branch after merging with new sockets

Chupakabra0 committed:
Update: Linux working build

Chupakabra0 committed:
Update: removed unneccesary macro

Chupakabra0 committed:
Added linux libs

Chupakabra0 committed:
Update: new libs for windows

Chupakabra0 committed:
Fix: removed logger param from one macros

Chupakabra0 committed:
Delete BackEndMonitoringSockets.lib

Chupakabra0 committed:
Merge branch 'logger' into Dev

VytalyGorbatov committed:
Fixed ThreadPool reaction on stop event; Added logger to ThreadPool; Beutify CService

VytalyGorbatov committed:
Linked with hardware monitoring, added much more logging, etc.

VytalyGorbatov committed:
Fixed data sending bug

Crowy-o committed:
Part loading json file (#33)

* Merged HardwareMonitoring branch with Dev branch

* Moved all OS-related code of processes monitoring

Moved all OS-related code of processes monitoring int PlatformUtils

* Renamed:
1. CHardwareInfoLifeCycle -> IHardwareInfoMonitoring
2. ThreadLifeCycle -> StartMonitoring
3. CLogicalDiskStatus -> CLogicalDiskInfo
4. CalculateAsCountType -> ConvertToCountType
5. CLogicalDiskStatusLifeCycle -> CLogicalDiskInfoMonitoring
6. CProcessesMonitoringLifeCycle -> CProcessesInfoMonitoring
7. CProcess -> CProcessInfo
8. ComputeCpuUsage -> CountCpuUsage
Changes:
1. Small changes in order to avoid implicit conversion
2. Avoid warning about no using nodiscard
3. Fromatted to code convetion
4. All constructor's implementetion moved to source file
5. API function from CContainerOfLogicalDisk moved to PlatformUtils
6. Solved bug with using pointer to nullptr
7. ClogicaldiskInfoMonitioring now using move for Specification
8. ConvertToCountType now implemented in Utils.h. Code fixed according to new changes

* Fixed issue with processes monitoring

* BackendMonitoringServer.cpp edited according to Dev branch

* All fields of HardwareMonitoring classes was reordered according to c++ alignement rules.

* Main change:
1. Now, instead of rewriting whole file, we add new data in .json file
2. Some changed according to new feature
2.1 Some new function in Utils namespace
2.2 Data saving only once during 1 cycle in meth StartMoniring()

* Fixed initialization of bool variables

* 1. Added method CloneObject to class CHardwareSpecification
2. Instead of inheritance, now using composition for CCointainerOfLogicalDisk and CCointainerOfProcesses CHardwareStatusSpecification
3. RoundToDecimal moved to Utils
3.1 Changed CLogicalDiskInfo and CProcessInfo according to new changes
4. Updating information cycle inside CLogicalInfoMonitoring moved into CContainerLogical disk in order to avoid modifing by another class
5. Small changes about code style

* Restored some changes after previous commit

* Changed param of fucn TryCreateFileIfNotExist to const ref instead of ref

* Solved bug with endless loop if PC containe floppy disk

* Small update

* Changes according pull-request

Co-authored-by: VytalyGorbatov <75737862+VytalyGorbatov@users.noreply.github.com>
Co-authored-by: VytalyGorbatov <vytalygorbatovwork@gmail.com>

Crowy-o committed:
Logical disks linux update (#36)

* Merged HardwareMonitoring branch with Dev branch

* Moved all OS-related code of processes monitoring

Moved all OS-related code of processes monitoring int PlatformUtils

* Renamed:
1. CHardwareInfoLifeCycle -> IHardwareInfoMonitoring
2. ThreadLifeCycle -> StartMonitoring
3. CLogicalDiskStatus -> CLogicalDiskInfo
4. CalculateAsCountType -> ConvertToCountType
5. CLogicalDiskStatusLifeCycle -> CLogicalDiskInfoMonitoring
6. CProcessesMonitoringLifeCycle -> CProcessesInfoMonitoring
7. CProcess -> CProcessInfo
8. ComputeCpuUsage -> CountCpuUsage
Changes:
1. Small changes in order to avoid implicit conversion
2. Avoid warning about no using nodiscard
3. Fromatted to code convetion
4. All constructor's implementetion moved to source file
5. API function from CContainerOfLogicalDisk moved to PlatformUtils
6. Solved bug with using pointer to nullptr
7. ClogicaldiskInfoMonitioring now using move for Specification
8. ConvertToCountType now implemented in Utils.h. Code fixed according to new changes

* Fixed issue with processes monitoring

* BackendMonitoringServer.cpp edited according to Dev branch

* All fields of HardwareMonitoring classes was reordered according to c++ alignement rules.

* Fixed initialization of bool variables

* 1. Added method CloneObject to class CHardwareSpecification
2. Instead of inheritance, now using composition for CCointainerOfLogicalDisk and CCointainerOfProcesses CHardwareStatusSpecification
3. RoundToDecimal moved to Utils
3.1 Changed CLogicalDiskInfo and CProcessInfo according to new changes
4. Updating information cycle inside CLogicalInfoMonitoring moved into CContainerLogical disk in order to avoid modifing by another class
5. Small changes about code style

* Restored some changes after previous commit

* Changed types of containers

Changed type of PID container to vector;
Changed type of Processes container to list.

* Solved bug with endless loop if PC has a floppy disk

* Reworked CContainerOfLogicalDisks according to cross-platform modification

* Changed according to pull-request

* Merged Dev into current

* Changes according to pull-request

Co-authored-by: VytalyGorbatov <75737862+VytalyGorbatov@users.noreply.github.com>
Co-authored-by: VytalyGorbatov <vytalygorbatovwork@gmail.com>

Serwlashev committed:
Fixed problems with linux, now sockets can work on ubuntu

KokhanO committed:
Added Configurations (#39)

* Added pugixml lib

* Added XMLParser

* Fixed all bugs

* Added strategy pattern in CConfiguration class

added common interface for parsers,
added Configuration class with strategy pattern usage,
Fixed small logic and code style issues.

example of Configuration usage:
std::unique_ptr<IParser> p = std::make_unique<CXMLParser>();
    p->Initialize("D:\\xgconsole.xml");
    CConfiguration c(std::move(p));
    std::string ipaddress;
    if (c.TryToGetConfiguration("//root/Server/ipaddress", ipaddress))
    {
        std::cout << ipaddress << std::endl;
    }

* Added some verifications of transfered parameters

added verificaions in XMLParser, whether pointer to file exists, if Parcer wasn't able to find data with given data path and if data is equal to enable of disable

* Changed all architecture

the only solution that has come in a long time. Reworked all logic. Now returning structures with needed data. A lot of hardcode is present. If logic is ok, next task is to logg errors and exceptions. Also pugixml returns 38 warnings!

* Added one read method which reads all configurations, fixed bugs

std::unique_ptr<IParser> p = std::move(std::make_unique<CXMLParser>());
    p->Initialize("D:\\xgconsole.xml");
    CConfiguration config(std::move(p));
    config.ReadConfigFromFile();

example of Configuration class usage.
All possible nullpointers exception prevented, I do believe.

* Data Reader

* Added Settings classes

* A little start for great things...

* small fixes

* consts and explicits

* Added filters for Settings classes

* Code conventions

Co-authored-by: Chupakabra0 <rootalfa4@gmail.com>

Crowy-o committed:
Small update

VytalyGorbatov committed:
Linux implimentation of processes monitoring (#38)

* Added Linux realisation of processes monitoring && Added utils && Fixed minor issues

* Fixed according to pull request #38 && minor bug fixes

Serwlashev committed:
Sockets after marging with new branch, added logging to the Acceptor and CAcceptorWrapper

VytalyGorbatov committed:
Client-server communication reworked

Serwlashev committed:
added debugger to the server, added function to checking error from the client and server

Serwlashev committed:
Fixed few bugs on linux

VytalyGorbatov committed:
Fixed client reaction on discconect

Serwlashev committed:
Delete proc.json

Serwlashev committed:
Delete disk.json

Serwlashev committed:
Merge pull request #42 from ITA-Dnipro/Sockets

Sockets

Crowy-o committed:
Changes:
1. Added fucntion for converting int from xml to EMemoryCoinvertType
2. CService now use XML parser. Need some checks

VytalyGorbatov committed:
XML Config Fix

Crowy-o committed:
Added log level debug for all functions, added trace log level for creating local variables

Crowy-o committed:
Recovered main

Serwlashev committed:
Fixed problem with unexpected disconnected clients on windows

Serwlashev committed:
Fixing classes according to codestyle, added logging to the server

Serwlashev committed:
Fixed problem with client view

Crowy-o committed:
Fixed bug with logger inn Service and hddSettings (#55)

KokhanO committed:
Xml util (#53)

* Added pugixml lib

* Added XMLParser

* Fixed all bugs

* Added strategy pattern in CConfiguration class

added common interface for parsers,
added Configuration class with strategy pattern usage,
Fixed small logic and code style issues.

example of Configuration usage:
std::unique_ptr<IParser> p = std::make_unique<CXMLParser>();
    p->Initialize("D:\\xgconsole.xml");
    CConfiguration c(std::move(p));
    std::string ipaddress;
    if (c.TryToGetConfiguration("//root/Server/ipaddress", ipaddress))
    {
        std::cout << ipaddress << std::endl;
    }

* Added some verifications of transfered parameters

added verificaions in XMLParser, whether pointer to file exists, if Parcer wasn't able to find data with given data path and if data is equal to enable of disable

* Changed all architecture

the only solution that has come in a long time. Reworked all logic. Now returning structures with needed data. A lot of hardcode is present. If logic is ok, next task is to logg errors and exceptions. Also pugixml returns 38 warnings!

* Added one read method which reads all configurations, fixed bugs

std::unique_ptr<IParser> p = std::move(std::make_unique<CXMLParser>());
    p->Initialize("D:\\xgconsole.xml");
    CConfiguration config(std::move(p));
    config.ReadConfigFromFile();

example of Configuration class usage.
All possible nullpointers exception prevented, I do believe.

* Data Reader

* Added Settings classes

* A little start for great things...

* small fixes

* consts and explicits

* Added filters for Settings classes

* Code conventions

* Added interfaces

* Added time in hdd and processes

* fixed mistakes

* resolved conflicts

Co-authored-by: Chupakabra0 <rootalfa4@gmail.com>

Crowy-o committed:
Solved bbug with GetDate

Crowy-o committed:
Merge branch 'Dev' of https://github.com/ITA-Dnipro/BackEndMonitoring into Dev

Crowy-o committed:
Resetted to working version.
Solved bug with date

Crowy-o committed:
Creating

Crowy-o committed:
Check is xml correct

KokhanO committed:
fixed update

KokhanO committed:
Merge branch 'Dev-Fix-XML' into Dev

akolechko committed:
Added second logger for service, fixed XML parser

akolechko committed:
Merge pull request #57 from ITA-Dnipro/Service-logger

Added second logger for service, fixed XML parser

KokhanO committed:
Added time config reading

akolechko committed:
Removed absolute paths

akolechko committed:
Merge pull request #58 from ITA-Dnipro/remove-absolute-paths

Removed absolute paths

Serwlashev committed:
Redo client, change logic for receiving data, now clients write info to the file as well as to the console. Fixed sending and receiving data

VytalyGorbatov committed:
Merged Dev into current branch

akolechko committed:
Merge pull request #54 from ITA-Dnipro/Sockets

Fixed problem with unexpected disconnected clients on windows
